src/R4Mvc.Tools/Services/ControllerGeneratorService.cs

```diff
@@ -121,7 +122,27 @@ public ClassDeclarationSyntax GeneratePartialController(ControllerDefinition con
                     .WithModifiers(SyntaxKind.PublicKeyword)
                     .WithGeneratedNonUserCodeAttributes()
                     .ForEach(controllerMethodNames, (c, m) => c
-                        .WithStringField(m, m, SyntaxKind.PublicKeyword, SyntaxKind.ConstKeyword)));
+                        .WithStringField(m, m, SyntaxKind.PublicKeyword, SyntaxKind.ConstKeyword)))
+                /* [GeneratedCode, DebuggerNonUserCode]
+                 * static readonly ActionParamsClass_LogOn s_params_LogOn = new ActionParamsClass_LogOn();
+
+                 * [GeneratedCode, DebuggerNonUserCode]
+                 * public ActionParamsClass_LogOn LogOnParams { get { return s_params_LogOn; } }
+
+                 * [GeneratedCode, DebuggerNonUserCode]
+                 * public class ActionParamsClass_LogOn
+                 * {
+                 *     public readonly string returnUrl = "returnUrl";
+                 *     public readonly string model = "model";
+                 * }
+                */
+                .ForEach(methods.Where(m => m.Parameters.Any()), (c, m) => c
```

Comment:
Parameter generation should be optional (as in the original T4MVC), which would be decided from the settings json file with the name 'GenerateParamsForActionMethods'.